Sources/SKCore/BuildServerBuildSystem.swift

```diff
@@ -78,12 +87,36 @@ public final class BuildServerBuildSystem {
     let handler = BuildServerHandler()
     let buildServer = try makeJSONRPCBuildServer(client: handler, serverPath: serverPath, serverFlags: flags)
     let response = try buildServer.sendSync(initializeRequest)
+    buildServer.send(InitializedBuildNotification())
     log("initialized build server \(response.displayName)")
+
+    // see if index store was set as part of the server metadata
+    if let indexStorePath = readReponseDataKey(data: response.data, key: "indexStorePath") {
+      self.indexStorePath = AbsolutePath(indexStorePath, relativeTo: self.projectRoot)
+    }
     self.buildServer = buildServer
     self.handler = handler
   }
 }
 
+private func readReponseDataKey(data: LSPAny?, key: String) -> String? {
+  switch data {
+  case .dictionary(let dataDict):
```

Comment:
You can simplify this code a bit with `if case` pattern matching.

```swift
if case .dictionary(let dataDict)? = data,
   case .string(let stringVal)? = dataDict[key]
{
  return stringVal
}

return nil
```

Reply:
Was struggling with this one, knew there had to be a swiftier way to do this.